src/test/java/org/noteskeeper/server/GreetingServiceTest.java

```diff
@@ -0,0 +1,26 @@
+package org.noteskeeper.server;
+
+/**
+ * Created by ioan1 on 2/5/2017.
+ */
+
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+
+public class GreetingServiceTest {
+
+    private GreetingService service = new GreetingService();
+
+    @Test
+    public void greetGreetsAnonymousWhenPassedNullOrEmptyName() {
+        String expectedGreeting = "Hello, Anonymous";
+        assertEquals("",expectedGreeting, service.greet(null));
```

Comment:
you might omit empty assert message => assertEquals(expectedGreeting, service.great(null))